pallets/thea/src/lib.rs

```diff
@@ -0,0 +1,534 @@
+				payload.network_id,
+				payload.who,
+				payload.asset_id,
+				payload.amount,
+				payload.tx_hash,
+			));
+			Ok(())
+		}
+
+		/// Manually claim an approved deposit
+		///
+		/// # Parameters
+		///
+		/// * `origin`: User
+		/// * `num_deposits`: Number of deposits to claim from available deposits,
+		/// (it's used to parametrise the weight of this extrinsic)
+		#[pallet::weight(1000)]
+		pub fn claim_deposit(origin: OriginFor<T>, num_deposits: u32) -> DispatchResult {
+			let user = ensure_signed(origin)?;
+
+			if let Some(mut deposits) = <ApprovedDeposits<T>>::get(&user) {
+				let length: u32 = if deposits.len().saturated_into::<u32>() <= num_deposits {
+					deposits.len().saturated_into()
+				} else {
+					num_deposits
+				}
+				.saturated_into();
+
+				for _ in 0..length {
+					if let Some(deposit) = deposits.pop() {
+						if let Err(err) = Self::execute_deposit(deposit, &user) {
+							// Force push is fine as it will have the capacity.
+							deposits.force_push(deposit);
+							// Save it back on failure
+							<ApprovedDeposits<T>>::insert(&user, deposits.clone());
+							return Err(err)
+						}
+					} else {
+						break
+					}
+				}
+
+				if !deposits.is_empty() {
+					// If pending deposits are available, save it back
+					<ApprovedDeposits<T>>::insert(&user, deposits)
+				} else {
+					<AccountWithPendingDeposits<T>>::mutate(|accounts| accounts.remove(&user));
+				}
+			}
+
+			Ok(())
+		}
+
+		/// Extrinsic to update withdrawal completion status by relayer
+		///
+		/// # Parameters
+		///
+		/// * `origin`: Any relayer
+		/// * `withdrawal_nonce`: Withdrawal Nonce
+		/// * `network`: Network id
+		/// * `tx_hash`: Vec<u8>
+		/// * `bit_map`: Bitmap of Thea relayers
+		/// * `bls_signature`: BLS signature of relayers
+		#[pallet::weight(1000)]
+		pub fn batch_withdrawal_complete(
+			origin: OriginFor<T>,
+			withdrawal_nonce: u32,
+			network: Network,
+			tx_hash: sp_core::H256,
+			_bit_map: u128,
+			_bls_signature: [u8; 96],
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			// TODO: Something to verify BLS signature
+			<ReadyWithdrawls<T>>::take(network, withdrawal_nonce);
+
+			Self::deposit_event(Event::<T>::WithdrawalExecuted(withdrawal_nonce, network, tx_hash));
+			Ok(())
+		}
+
+		/// Initiate the withdraw for user
+		///
+		/// # Parameters
+		///
+		/// * `origin`: User
+		/// * `asset_id`: Asset id
+		/// * `amount`: Amount of asset to withdraw
+		#[pallet::weight(1000)]
+		pub fn withdraw(
+			origin: OriginFor<T>,
+			asset_id: u128,
+			amount: u128,
+			beneficiary: Vec<u8>,
+			pay_for_remaining: bool,
+		) -> DispatchResult {
+			let user = ensure_signed(origin)?;
+			// Put a soft limit of size of beneficiary vector to avoid spam
+			ensure!(beneficiary.len() <= 100, Error::<T>::BeneficiaryTooLong);
+
+			// Find native network of this asset
+			#[allow(clippy::unnecessary_lazy_evaluations)]
+			// TODO: Remove once merged to asset-handler
```

Comment:
Please elaborate on this comment

Reply:
This will be refactored when work on withdrawal so not fixing clippy suggestion for now